firefox-ios/Client/Frontend/Browser/SearchViewController.swift

```diff
@@ -133,13 +133,25 @@ class SearchViewController: SiteTableViewController,
     var searchFeature: FeatureHolder<Search>
     static var userAgent: String?
 
+    private var bookmarkSites: [Site] {
+        data.compactMap { $0 }
+            .filter { $0.bookmarked == true }
+    }
+
+    private var historySites: [Site] {
+        data.compactMap { $0 }
+            .filter { $0.bookmarked == false }
+    }
+
     var hasFirefoxSuggestions: Bool {
         let dataCount = data.count
-        return dataCount != 0
-            || !filteredOpenedTabs.isEmpty
-            || !filteredRemoteClientTabs.isEmpty
-            || !searchHighlights.isEmpty
-            || !firefoxSuggestions.isEmpty
+        return dataCount != 0 &&
+                (model.shouldShowBookmarksSuggestions ||
+                model.shouldShowBrowsingHistorySuggestions)
+                || !filteredOpenedTabs.isEmpty
```

Comment:
```suggestion
        return dataCount != 0
                && (model.shouldShowBookmarksSuggestions || model.shouldShowBrowsingHistorySuggestions)
                || !filteredOpenedTabs.isEmpty
```

nit: keeps the formatting similar throughout the expression